framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -923,6 +1066,50 @@ private AnnotationMirror createNumberAnnotationMirror(List<Number> values) {
                 "ValueAnnotatedTypeFactory: unexpected class: " + first.getClass());
     }
 
+    public AnnotationMirror createIntRangeAnnotation(Range range) {
```

Comment:
By query method, you mean a function like `Range.IsNothing()`? If this is the case, it answers my first question above.